### DISCUSSION TITLE
forecasting using LightGBM

### DISCUSSION THREAD

**Discussion by nyk2001** [alan-turing-institute/sktime#3783] [Ideas]
I have used ARIMA and Prophet algorithms for multivariate forecasting, which works fine.  You can see the plot with Prophet as follows:

![image](https://user-images.githubusercontent.com/3079053/201594068-3b15afa1-ddc5-4e1c-b866-d2deab379f45.png)

Now, I am trying to use LightGBM to make predictions and it does not work. For instance, I am getting the following output:

![image](https://user-images.githubusercontent.com/3079053/201594207-89d4df80-c6e7-4b31-a66d-c490b3ac42e7.png)

Please help me to resolve the issue. My code is as follows: 

`
# Load data
df = sm.datasets.macrodata.load_pandas()['data']
df['q_date'] = df.apply(lambda x: str(int(x['year'])) + '-Q' + str(int(x['quarter'])), axis=1)
df['date'] = pd.PeriodIndex(df['q_date'], freq='Q').to_timestamp()
df.loc[:, 'realinv_lagged'] = df.loc[:, 'realinv'].shift()
df[['realinv_lagged']] = df[['realinv_lagged']].fillna(method='backfill')

# Split into x and y
y = df['realgdp']
X = df[['realinv_lagged','cpi','m1']] #could be more features

# Split dataset
y_train, y_test, X_train, X_test = temporal_train_test_split(y, X, test_size=30)

# Create LGBM regressor
regressor = lgb.LGBMRegressor()
forecaster = make_reduction(regressor, window_length=10, strategy="recursive", scitype="tabular-regressor")
    
# Define horizon
fh_abs = ForecastingHorizon(y_test.index, is_relative=False) 

# Fit model
forecaster.fit(y_train, X_train)

# Make predictions
y_pred = forecaster.predict(X=X_test, fh=fh_abs)

# Plot preditcions
plot_series(y_train, y_test, y_pred, labels=["y_train", "y_test", "y_pred"])
`

**Comment by nyk2001** [comment]
Please respond as I need to solve this problem as soon as possible. I tried using Linear Regression from sklearn and that works ok. I also tried using GradientBooster but that does not work

**Reply by AugustComte** [reply]
The problem here is caused by the fact that you are [using a tree based algorithm to predict values outside of training](https://stats.stackexchange.com/questions/304962/is-is-possible-for-a-gradient-boosting-regression-to-predict-values-outside-of-t  ), both prophet and ARIMA use some aspect of linear regression which allow them to predict never before seen values. 

I believe the answer is to use [linear trees](https://github.com/microsoft/LightGBM/pull/3299) for gbm models, see link for lgbm details. Not sure if passing in params is best way to add with sktime though.

**Comment by fkiraly** [comment]
I think you posted this twice, @nyk2001?

You have received some answers here:
https://github.com/sktime/sktime/issues/3784

Summary:
* what @AugustComte said, tree ensembles do not extrapolate
* you can help them extrapolate by applying a transformation/inversion step that makes the time series more stationary, e.g., differencer (see example in #3784)
